### ISSUE TITLE
OpenLens 6.3.0 on macOS does not show pods and config/network/storage sections are empty, all work fine in 6.2.6.

### ISSUE THREAD

**Issue by dmarjanovic** [kontena/lens#6858]
**Describe the bug**
OpenLens 6.3.0 on macOS does not show pods and config/network/storage sections are empty, all work fine in 6.2.6.

**To Reproduce**
1. git checkout v6.3.0 # or master d6531f2
2. make clean && make dev
3. When window appears, connect to a namespace (there's already configured single namespace in Namespace/Accessible Namespaces)
4. Expand 'Workloads'
5. Do not see Pods, but can see Deployments, DaemonSets, StatefulSets, ReplicaSets, Jobs, CronJobs
6. Scroll down to 'Config' and expand
7. Do not see ConfigMaps, Secrets, Resource Quotas, Limit Ranges, HPA, Pod Disruption Budgets
8. Scroll down to 'Network' and expand
9. Do not see Services, Endpoints, Ingresses, Network Policies but can only see Port Forwarding
10. Scroll down to 'Storage'
11. Can't expand as it's empty (can't see Persistent Volume Claims')

**Expected behavior**
5. I see Pods in 'Workloads' section
7. I see ConfigMaps, Secrets, Resource Quotas, Limit Ranges, HPA, Pod Disruption Budgets in 'Config' section
9. I see Services, Endpoints, Ingresses, Network Policies in 'Network' section
11. I see Persistent Volume Claims in 'Storage' section

**Screenshots**
Wrong one from OpenLens 6.3.0:
![Screenshot 2023-01-02 at 16 21 36](https://user-images.githubusercontent.com/1816763/210252004-49b4b29f-038a-4627-83f1-3324cfd8cd60.png)

Correct one from OpenLens 6.2.6:
![Screenshot 2023-01-02 at 16 14 32](https://user-images.githubusercontent.com/1816763/210252061-ef8863ee-84f4-4e7a-87e0-e45fc000575e.png)

**Environment (please complete the following information):**
- Lens Version: 6.3.0
- OS: macOS
- Installation method (e.g. snap or AppImage in Linux): make clean && make dev # from github tag

**Kubeconfig:**
```
apiVersion: v1
clusters:
- cluster:
    server: https://api-kube.***
  name: cluster1
...
contexts:
- context:
    cluster: cluster1
    namespace: ns1
    user: some-user
  name: cluster1-ns1
...
current-context: cluster1-ns1
kind: Config
preferences: {}
users:
- name: some-user
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - oidc-login
      - get-token
      - --oidc-issuer-url=https://***
      - --oidc-client-id=api-kube
      - --oidc-client-secret=***
      - --oidc-extra-scope=groups
      command: kubectl
      env: null
      interactiveMode: IfAvailable
      provideClusterInfo: false
```

**Comment by Nokel81** [comment]
What sort of kube distro is this cluster for?

**Comment by Fantaztig** [comment]
Same (or even worse) issue for me.
In my case I cannot see any resources besides CRDs which work fine.

Were running on VMWare Tanzu, kubernetes version 1.22.9

I have cluster admin permissions inside the cluster and can get/describe all resources via the CLI.

Installed lens 6.2.6 and was upgraded to 6.3.0, in the old version everything worked as expected.

**Comment by Nokel81** [comment]
Fixed by https://github.com/lensapp/lens/pull/6880

**Comment by dmarjanovic** [comment]
@Nokel81 

> What sort of kube distro is this cluster for?

We're with aws eks, if that helps.

> Fixed by #6880

The #6880 contribution feels much more stable (faster) than 3.6.0 but the problem (and screenshots) described in this issue are still not solved with it. I've verified with latest commit d34a13fad2 in `farodin91:set-correct-group-in-rbac` remote:branch

Any other ideas? Tnx

**Comment by Nokel81** [comment]
It certainly feels like an RBAC related issue. What is the response for the following command?

```
kubectl create -f - -o yaml << EOF
apiVersion: authorization.k8s.io/v1
kind: SelfSubjectRulesReview
spec:
  namespace: default
EOF
```

**Comment by dmarjanovic** [comment]
@Nokel81 I can't reveal config details sorry. Is there something specific you're looking for? Or let me know if there's another way to help diagnosing this issue. Thank you

**Comment by Nokel81** [comment]
@dmarjanovic Okay fair enough. How about this...

if you run `kubectl get --raw /api` you should get back a JSON object that has a field called `versions`. What is its value? On GKE and minikube it is `["v1"]`

**Comment by gfarcas** [comment]
I have the same problem, 6.2.5 or k9s works though

**Comment by Nokel81** [comment]
@gfarcas Can you run the command that I asked @dmarjanovic to run?

**Comment by HuguesJ** [comment]
Same issue on Windows, with an AKS cluster. I run the command from @dmarjanovic and got the same result than him.

**Comment by Nokel81** [comment]
Can you please also run `kubectl get --raw /api/v1` then? Do you get a list that includes `pods`?

**Comment by HuguesJ** [comment]
@Nokel81 Not sure what you mean by pods. 
`kubectl get --raw /api/v1` returns a json object with some elements with `pods` in their name like `pods`,`pods/attach`, `pods/binding` , it doesn't return the list of all pods.

**Comment by Nokel81** [comment]
No that is what I meant and expected you to get. Hmmm.... 🤔

**Comment by HuguesJ** [comment]
@Nokel81 Interesting enough, this problem happens only in one cluster (AKS) out of 4 (AKS and GKE).

**Comment by Nokel81** [comment]
@HuguesJ Thanks for the info, will investigate against an AKS cluster

**Comment by Nokel81** [comment]
Here is some more debugging that would be helpful

1. Does `kubectl get namespaces` succeed? (I assume it does)
2. Running the following

```
kubectl create -f - -o yaml << EOF
apiVersion: authorization.k8s.io/v1
kind: SelfSubjectRulesReview
spec:
  namespace: default
EOF
```

both succeeds and returns something like:

```yaml
apiVersion: authorization.k8s.io/v1
kind: SelfSubjectRulesReview
metadata:
  creationTimestamp: null
spec: {}
status:
  incomplete: false
  nonResourceRules:
  - nonResourceURLs:
    - /healthz
    - /livez
    - /readyz
    - /version
    - /version/
    verbs:
    - get
  - nonResourceURLs:
    - '*'
    verbs:
    - '*'
  - nonResourceURLs:
    - /api
    - /api/*
    - /apis
    - /apis/*
    - /healthz
    - /livez
    - /openapi
    - /openapi/*
    - /readyz
    - /version
    - /version/
    verbs:
    - get
  resourceRules:
  - apiGroups:
    - authorization.k8s.io
    resources:
    - selfsubjectaccessreviews
    - selfsubjectrulesreviews
    verbs:
    - create
  - apiGroups:
    - '*'
    resources:
    - '*'
    verbs:
    - '*'
```

Note the things I am interested about is the `.status.resourceRules` part, is there an entry similar to:

```yaml
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
```

If not, are the fields `apiGroups` and `resources` empty, or maybe are they missing entirely?

**Comment by dmarjanovic** [comment]
@Nokel81 sry for late answer, I'm unable to iterate quickly.

Re.
> if you run `kubectl get --raw /api` you should get back a JSON object that has a field called `versions`. What is its value? On GKE and minikube it is `["v1"]`

```
$ kubectl get --raw /api
{"kind":"APIVersions","versions":["v1"]...
```

Re.
> 1. Does `kubectl get namespaces` succeed? (I assume it does)

Yes, there are multiple namespaces and the same regression is present in all.

> 2. Running the following
> kubectl create -f - -o yaml << EOF...

In `.status.resourceRules` in the output there's not the below part you expect to be shown:
```
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
```

Instead, I can see (among few other info that I had to remove) 20 times repeated this part (btw not sure why it's redundant):
```
- apiGroups:
    - ""
    resources:
    - nodes
    - namespaces
    verbs:
    - get
    - list
    - watch
  - apiGroups:
    - metrics.k8s.io
    resources:
    - nodes
    verbs:
    - get
    - list
    - watch
  - apiGroups:
    - apiextensions.k8s.io
    resources:
    - customresourcedefinitions
    verbs:
    - get
    - list
    - watch
```

In addition, I see:
```
- apiGroups:
  - policy
  resourceNames:
  - eks.privileged
  resources:
  - podsecuritypolicies
  verbs:
  - use
```
which is probably the way how to configure it in AWS EKS in idiomatic way but I am not aware of technical details except that part is also documented in https://docs.aws.amazon.com/eks/latest/userguide/pod-security-policy.html

Anyways, I would expect that some OpenLens code relevant to k8s resources discovery changed in 6.3.0 compared to 6.2.6, not aware of diff though. Any other hints on how to diagnose this further? Thank you for looking into this.

**Comment by Nokel81** [comment]
Yes there was a bug fix that did change the related code.

In the list is there something like:

```yaml
  - apiGroups:
    - ""
    resources:
    - pods
    verbs:
    - get
    - list
    - watch
```

?

**Comment by dmarjanovic** [comment]
@Nokel81 no, there's not such part. I assume EKS does it differently as mentioned with policy from my previous comment, but not sure.

**Comment by dmarjanovic** [comment]
@Nokel81 btw, there's one detail connected to resources discovery that might be or might not be connected, but we noticed even with older (working) OpenLens versions similar "buggy" behaviour of not seeing those resources (including pods) mentioned in this ticket when connecting to some context for the first time. Only after namespace name is explicitly set (see screenshot) then disconnect+connect would show all the resources in this namespace - this "workaround" worked fine until 6.2.6 but not now in 6.3.0, not sure if it's connected though. We have permission scheme configured per namespace, not cluster.

<img width="1399" alt="Screenshot 2023-01-10 at 14 09 04" src="https://user-images.githubusercontent.com/1816763/211560267-99d17384-f63a-4730-845e-33e5cc7862eb.png">

**Comment by Nokel81** [comment]
So that list is for when the user does not have permissions to `list namespaces`. We do still read that list, but the change that is present in 6.3.0 attempts to fix the bug where only the first 10 namespaces are checked for these permissions.

**Comment by dmarjanovic** [comment]
@Nokel81 is it possible to pin point to specific PR causing this behaviour, maybe #6657?

**Comment by Nokel81** [comment]
That PR fixes issues with https://github.com/lensapp/lens/pull/6614 to support "incomplete" responses like those which GKE return.

**Comment by kienhoefr** [comment]
> Yes there was a bug fix that did change the related code.
> 
> In the list is there something like:
> 
> ```yaml
>   - apiGroups:
>     - ""
>     resources:
>     - pods
>     verbs:
>     - get
>     - list
>     - watch
> ```
> 
> ?

@Nokel81 I'm working with AWS EKS (using OpenLens 6.3.0) and I have the same issue.
When I do a `SelfSubjectRulesReview` I get (among other things):

```yaml
- apiGroups:
    - ""
    resources:
    [...]
    - pods
    [...]
    - services
    [...]
    verbs:
    - get
    - list
    - watch
```

Hope this helps.

**Comment by Nokel81** [comment]
Yes that helps a lot, thanks

**Comment by dmarjanovic** [comment]
@Nokel81 thank you for the fix.

Btw #6900 only partially fixes this issue for us in a way that after the fix is applied the issue is not reproducible for some namespaces and still persists for the other ones. To be more precisely: the "working" or "non-working" namespaces are now different after OpenLens app restart meaning on 1st start the namespace A is "working" and B is "non-working" but on 2nd run A may become "non-working" and B "working" etc. Weird behaviour. Also, I didn't see anything weird (or different) in logs comparing to v6.2.6.

**Comment by Nokel81** [comment]
By working what do you mean exactly? We attempt to list all the namespaces and then attempt to compute which resources are allowed to be listed in at least one of those namespaces. What namespace you select in the UI shouldn't matter (except for the "Accessible Namespaces" setting which just overrides the list namespace step above).

**Comment by dmarjanovic** [comment]
@Nokel81 sorry for using confusing terminology. I built and run app from master c361852dd2 (6.4.0-alpha.3). All below screens were from single run, no restarts performed.

We use single ~/.kube/config file with single eks cluster, many namespaces, 1 context per namespace. It appears in Lens Catalog/Clusters as 11 "clusters" but matches basically 11 contexts. Btw, it says Distro "eks" but not for all contexts, not sure why.
<img width="1441" alt="02_openlens_6 4 0-alpha 3_11_clusters_connected" src="https://user-images.githubusercontent.com/1816763/212421480-034511e4-12d7-4672-9a7b-95df43ff8fcd.png">

Steps I did: was clicking from the top starting with the 1st pinned "cluster" (context) then clicked on 2nd and so on until the last 11th pinned "cluster". Some are "working" meaning pods **are** shown and the other ones are "failed" meaning **no** pods are shown. See first 11 screenshots:
<img width="332" alt="03_1 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212423460-d1e232f7-e052-4e41-9279-809c515c78d7.png">
<img width="548" alt="04_2 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212423464-7680f9b2-791c-4807-a58d-52f19425affd.png">
<img width="346" alt="05_3 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212423468-0cf18f4e-0779-4549-9436-022c9db2af00.png">
<img width="331" alt="06_4 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212423471-fb0c4627-3eca-467f-88f6-a4b164af10c8.png">
<img width="336" alt="07_5 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212423474-a74604d5-25ae-4306-93eb-c9f78fbdd37a.png">
<img width="283" alt="07_6 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212423477-96bd8f0e-95bd-4102-8c75-e74322d6ec5f.png">
<img width="278" alt="08_7 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212423480-685a881e-0a5b-49bb-a0bc-451ee1d68f14.png">
<img width="278" alt="09_8 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212423484-4d963797-5cdb-4d2b-8655-f814a333ce9c.png">
<img width="279" alt="10_9 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212423485-f1e081e4-ab8d-4741-a631-00044e2b08bc.png">
<img width="277" alt="11_10 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212423488-32416627-3230-44b6-8033-9236ac28c0b1.png">
<img width="279" alt="12_11 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212423490-e4140966-71ca-4a58-9175-571d3252cf80.png">

- Then I picked and disconnected only one "failed" "cluster" - I was thinking that reconnecting might solve the issue
- OpenLens screen got all empty, no buttons or anything (like on screenshot below)
<img width="794" alt="Screenshot 2023-01-13 at 22 53 10" src="https://user-images.githubusercontent.com/1816763/212426149-45cc0450-a7ec-4e82-8fbd-763bdf9802e0.png">

- clicked cmd+r to reload
- UI appeared but it seemed like all the "clusters" were disconnected and had to be connected again, though there were green dots in pinned icons shown (looks like a bug)
- I was clicking first on those 5 previously "failed" "clusters" and they were now "working":
<img width="278" alt="13_9 cluster_ok_after_disconnect_ctrl-r_connect" src="https://user-images.githubusercontent.com/1816763/212424352-f833c982-23f8-4b7f-bf19-46e59e288e14.png">
<img width="329" alt="14_1 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424354-771d7c16-dea3-492c-82e2-7aff4c41569e.png">
<img width="336" alt="15_3 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424357-42dcc4c1-7eeb-489e-9fdb-6153754815e1.png">
<img width="280" alt="16_8 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424360-f11a18d3-b6bd-44d9-8ba9-237919be658a.png">
<img width="334" alt="18_5 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424366-668fe69a-9288-4c38-83fe-3fe4f48de452.png">

I started clicking on other pinned "clusters" that were "working" before but now they were changing from "working" to "failed" or from "working to failed to working" and even those that were "failed" then "working" started going to "failed" then some of them to "working" again without any meaningful pattern.
<img width="335" alt="17_4 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212424363-ad53d98f-e4ec-4558-a1c9-a7dd6b504fb8.png">
<img width="278" alt="19_8 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212425051-869b9c5b-d7da-47e4-b2de-5518441b5c7f.png">
<img width="275" alt="20_10 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212424974-06de2edc-fa00-466b-931c-b95440669919.png">
<img width="275" alt="21_9 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212424976-a9420e71-6326-4589-a6bb-a8ee8e2ad951.png">
<img width="275" alt="22_6 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212424978-af573863-bc82-4fc0-bbcb-8079c66e30b1.png">
<img width="331" alt="23_5 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212424981-3d6cbc7a-da4e-4239-876c-82ffc44fee31.png">
<img width="331" alt="24_1 cluster_fails" src="https://user-images.githubusercontent.com/1816763/212424985-3063ab8d-5cb4-4a03-800f-1b663d2d9fa5.png">
<img width="273" alt="25_8 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424991-4f0ea66f-7b5d-4e40-ba24-8d546f82754c.png">
<img width="276" alt="26_9 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424994-6fa56353-6b4b-4e3d-965e-9925098720c3.png">
<img width="274" alt="27_10 cluster_ok" src="https://user-images.githubusercontent.com/1816763/212424999-c3250ffa-2fef-4512-a2e5-049114d71839.png">

At this point I was clicking on all the 11 pinned "clusters" but their state was not changing any more.

Now I did restart and repeat clicking on 11 pinned "clusters" and now "working" and "failed" ones were different than in previous run. Totally random behaviour comparing to previous run. No pattern found. Looked to me like there was some race condition going on. While doing screenshots I did breaks to work on screen, not sure if that influenced behaviour too.

Note that all these namespaces (the 11 contexts) defined in ~/.kube/config do have pods and are shown from all the other tools correctly including k9s, kubectl, OpenLens 6.2.6.

**Comment by Nokel81** [comment]
Okay thanks.

One question. Do you have list namespace permissions for this Kube cluster?

**Comment by dmarjanovic** [comment]
@Nokel81 yes, I can list namespaces everywhere.

**Comment by dmarjanovic** [comment]
@Nokel81 not sure if anyone is still looking into this issue but shouldn't it be open as it's regression in 6.3.0 and still not fully resolved in 6.4.0-alpha.3 comparing to 6.2.6? Tnx

**Comment by Nokel81** [comment]
@dmarjanovic for 6.4.0-alpha.3 do you have "accessible namespaces" configured for those? That might cause this, though it should be stable. We plan to make this more resilient against network failures.

**Comment by dmarjanovic** [comment]
@Nokel81 yes, "accessible namespaces" was indeed configured for every "cluster". So workaround is to remove "accessible namespaces" from every "cluster" then issue is not reproducible any more for me. Tested with 6.4.0-alpha.3. Also, workaround "works" kind of with 6.3.0 too but that version is much less stable.

Should I report separate issue for "accessible namespaces" or you have it on your radar? Thank you

**Comment by Nokel81** [comment]
So that list is really only supposed to be fore when a "cluster" (or more specifically the user associated with the context within a Kubeconfig file) does not have `LIST namespace` permissions.

Namely, if you configure those then we only will look at those namespaces when determining the "resources to show". So it would only be a bug if the same cluster changes which resources it shows when reconnected to.

If IIRC that is indeed what you said was happening, right?

**Comment by djarami726** [comment]
Same issue on Windows - doesn't work on 6.3.0 nor 6.4.0-alpha.3.   I was not able to find 6.2.6 installer - only found 6.2.5 and it didn't work with that one either.   I don't see the cluster, nodes options and no items listed under overview.  This problem happens on one of my AKS clusters, I have several others where these items show up fine.  I'm able to do get pods -A and get ns -A.  

If you want me to try 6.2.6, please share a link for windows that I can try.

**Comment by dmarjanovic** [comment]
> So that list is really only supposed to be fore when a "cluster" (or more specifically the user associated with the context within a Kubeconfig file) does not have LIST namespace permissions.

@Nokel81 I see, thanks for the explanation. To keep it sane my colleagues and I will do more tests when next iteration is available via brew. Will then also re-check with my colleagues that do not have `LIST namespace` permission and see if it's blocking them or they experience issue (since for me not using "accessible namespaces" works just fine for now). Thank you

**Comment by vbabenkoru** [comment]
I got the same issue, 6.2.* is working fine. We're running EKS.

**Comment by tintranvan** [comment]
I am using EKS, for Service Account Role i have add one more ClusterRole and ClusterRoleBinding to allow Service Account who can get namespace

```
 kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: namespace-list-role
rules:
- apiGroups: [""]
  resources: ["namespaces"]
  verbs: ["get", "list", "watch"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: namespace-list-role
roleRef: 
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: namespace-list-role
subjects: # points to my ServiceAccount
- kind: Group
  name: ns-fullaccess-group
  apiGroup: rbac.authorization.k8s.io
```

Beside namespace permission for EKS Accounts with Role and RoleBinding
It works well for me.
